### PR TITLE
utils: add sort method in tvgArray

### PR DESCRIPTION
**Pull request by RuiwenTang**
This PR does the following things:
* Add a new method in tvg::Array to support sort with custom comparer

The tessellation code in #1531 use tvg::Array as much as possible, and the algorithm need to sort Point , Edge.
So I add the sort method in tvg::Array. This is a new method so there will be no impact on existing code.

### CODE REVIEW

File: src/utils/tvgArray.h [hunk 1]
```diff
@@ -121,6 +128,34 @@ struct Array
...
+            T       tmp = arr[low];
+            while (i < j) {
+                while (i < j && !COMPARE{}(arr[j], tmp)) --j;
+                // while (i < j && arr[j] >= tmp) --j;
+                if (i < j) {
+                    arr[i] = arr[j];
+                    ++i;
+                }
+                while (i < j && COMPARE{}(arr[i], tmp)) ++i;
+                // while (i < j && arr[i] <= tmp) ++i;
```
Comment (hermet) [inline src/utils/tvgArray.h:148]: does this comment necessary?

**Review by hermet — CHANGES_REQUESTED** [review]
Please remove the test-case. We don't add it for the internal functions. Only extern APIs would have it.  Otherwise, Looks good to me. 

Thanks. :-)

**Comment by RuiwenTang** [comment]
@hermet Please take a look again :)

**Review by hermet — APPROVED** [review]
Great, looks good to me. thanks!

**Comment by hermet** [comment]
Ah one more, could you please revise the commit message?

**Comment by RuiwenTang** [comment]
> Ah one more, could you please revise the commit message? Remove -> Add a associate unite test case to check the sort logical

Sorry about the commit meesage, I have removed them

**Pull request merged**